lib/linux_kernel/kthread.c

```diff
@@ -228,17 +228,21 @@ M0_INTERNAL int m0_thread_confine(struct m0_thread *q,
 		  put_task_struct(p);
 		*/
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0)
+#ifdef CONFIG_X86_64
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0)
 		cpumask_copy(&p->cpus_mask, cpuset);
-#else
+	#else
 		cpumask_copy(&p->cpus_allowed, cpuset);
-#endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
+	#endif
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
 		p->nr_cpus_allowed = nr_allowed;
-#else
+	#else
 		p->rt.nr_cpus_allowed = nr_allowed;
+	#endif
+#elif defined CONFIG_AARCH64 /*aarch64*/
+		cpumask_copy(&p->cpus_allowed, cpuset);
+		p->nr_cpus_allowed = nr_allowed;
 #endif
-
```

Comment:
This code seems to be kernel version specific rather than architecture specific. What kernel version did you use to test it with?

Reply:
I have added the linux kernel version if statement in aarch64  specific section.( Kernel version: 4.18.0)

Reply:
So I'd suggest simplifying the code like this:

```c
#if LINUX_VERSION_CODE < KERNEL_VERSION(4,18,0) || defined CONFIG_AARCH64
                cpumask_copy(&p->cpus_allowed, cpuset);
#else
                cpumask_copy(&p->cpus_mask, cpuset);
#endif
#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0)
                p->rt.nr_cpus_allowed = nr_allowed;
#else
                p->nr_cpus_allowed = nr_allowed;
#endif
```

Reply:
This issue is still unresolved.

Reply:
It is  difficult to read like this since it mixes  code  from lot of flag variables which is not recommended. The code that is  there is working properly let it be like that.

Reply:
Your change makes the code very complex and hard to read. Moreover, it duplicates the `p->nr_cpus_allowed = nr_allowed;` line.

Please, simplify it as suggested.